Src/Support/Google.Apis.Auth/OAuth2/ITokenAccessWithExtraInformation.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+Copyright 2019 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System.Collections.Generic;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Represents a token access that provides extra information 
+    /// that either describes the access token or is associated with it.
+    /// </summary>
+    public interface ITokenAccessWithExtraInformation : ITokenAccess
+    {
+        /// <summary>
+        /// Collection containing the extra information associated
+        /// with the access token.
+        /// </summary>
+        IReadOnlyDictionary<string, string> ExtraInformation { get; }
```

Comment:
alternatively, there could be an overload of GetAccessTokenForRequestAsync which would return a class will all the info needed (auth token and all the extra headers) or would populate the extra headers with e.g. a callback?

`Task<TokenWithExtraInfo> GetAccessTokenWithExtraInfoForRequestAsync(....)`

`Task<string> GetAccessTokenForRequestAsync(Action<object, string, string> addExtraInfoCallback, object callbackArg, string authUri = null, CancellationToken cancellationToken = default(CancellationToken));`

Reply:
Hmmm, I think I like the second one better. We'd still need to create the ITokenAccessWithExtra... interface, if we add anything to ITokenAccess, we would be breaking clients.

Reply:
Now that I'm cleaning up this code I've thought a little bit more about this and:

- I think that by having a callback we are complicating things a little for client code.
  - One such complication is the object callbackArg, 
  - another is that the callback can't be async (unless we add overloads), 
  - if client code needed to have a more complex logic around headers, say, only send A if B is also sent, then it would be hard.

@jtattermusch I suppose that both yout proposals are around having the token and the headers as a unit which I agree with, so I'll go with the overload.